flask/templating.py

```diff
@@ -101,7 +101,20 @@ def list_templates(self):
 
 
 def _render(template, context, app):
-    """Renders the template and fires the signal"""
+    """Renders the template and fires signals.
+
+    It is possible to override render template in the before_render_template signal.
+    It can be done only if exactly one receiver and it return not None result."""
+
+    brt_resp = before_render_template.send(app, template=template, context=context)
+    
+    overrides = [rv for _, rv in brt_resp if rv is not None]
```

Comment:
Similar to how handling errors from errorhandlers works, I think Flask should do more to avoid these kind of situations, for example by somehow not firing the signal a second time (no idea how), or by allowing the signal user to only modify the context instead of overriding the return value (as you mentioned in the original issue).